File: src/main/java/ar/edu/um/tesoreria/rest/extern/model/PreTurnoFacultad.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.extern.model;

import java.io.Serializable;
import java.time.OffsetDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PreTurnoFacultad implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 6030476707501270581L;

	protected Long uniqueId;
	protected Integer facultadId;
	protected Integer lectivoId;
	protected Integer geograficaId;
	protected Integer turnoId;
	protected String nombre;
	protected OffsetDateTime inicio;
	protected OffsetDateTime fin;

	public String getKey() {
		return this.facultadId + "." + this.lectivoId + "." + this.geograficaId + "." + this.turnoId;
	}

}

File: src/main/java/ar/edu/um/tesoreria/rest/extern/model/kotlin/PreTurnoFacultad.kt
package ar.edu.um.tesoreria.rest.extern.model.kotlin

import java.time.OffsetDateTime

data class PreTurnoFacultad(

    var uniqueId: Long? = null,
    var facultadId: Int? = null,
    var lectivoId: Int? = null,
    var geograficaId: Int? = null,
    var turnoId: Int? = null,
    var nombre: String? = null,
    var inicio: OffsetDateTime? = null,
    var fin: OffsetDateTime? = null,

    ) {
    fun getKey(): String? {
        return facultadId.toString() + "." + lectivoId + "." + geograficaId + "." + turnoId
    }

}
